cfme/common/provider.py

```diff
@@ -684,7 +685,7 @@ def exists(self):
     def wait_for_delete(self):
         try:
             provider_rest = self.appliance.rest_api.collections.providers.get(name=self.name)
-        except ValueError:
+        except (ValueError, APIException):  # if the record doesn't exist, APIException from 404
```

Comment:
Fix for `test_provider_crud()`  failures where the record is already deleted by the time we call wait_for_delete.